frame/system/src/lib.rs

```diff
@@ -711,13 +418,357 @@ decl_module! {
 		/// # </weight>
-		#[weight = (T::SystemWeightInfo::suicide(), DispatchClass::Operational)]
-		pub fn suicide(origin) {
+		#[pallet::weight((T::SystemWeightInfo::suicide(), DispatchClass::Operational))]
+		pub fn suicide(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin)?;
 			let account = Account::<T>::get(&who);
 			ensure!(account.refcount == 0, Error::<T>::NonZeroRefCount);
 			ensure!(account.data == T::AccountData::default(), Error::<T>::NonDefaultComposite);
 			Self::kill_account(&who);
+			Ok(().into())
+		}
+	}
+
+	/// Event for the System module.
+	#[pallet::event]
+	pub enum Event<T: Config> {
+		/// An extrinsic completed successfully. \[info\]
+		ExtrinsicSuccess(DispatchInfo),
+		/// An extrinsic failed. \[error, info\]
+		ExtrinsicFailed(DispatchError, DispatchInfo),
+		/// `:code` was updated.
+		CodeUpdated,
+		/// A new \[account\] was created.
+		NewAccount(T::AccountId),
+		/// An \[account\] was reaped.
+		KilledAccount(T::AccountId),
+	}
+
+	/// Old name generated by `decl_event`.
+	#[deprecated(note = "use `Event` instead")]
+	pub type RawEvent<T> = Event<T>;
+
+	/// Error for the System module
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The name of specification does not match between the current runtime
+		/// and the new runtime.
+		InvalidSpecName,
+		/// The specification version is not allowed to decrease between the current runtime
+		/// and the new runtime.
+		SpecVersionNeedsToIncrease,
+		/// Failed to extract the runtime version from the new runtime.
+		///
+		/// Either calling `Core_version` or decoding `RuntimeVersion` failed.
+		FailedToExtractRuntimeVersion,
+		/// Suicide called when the account has non-default composite data.
+		NonDefaultComposite,
+		/// There is a non-zero reference count preventing the account from being purged.
+		NonZeroRefCount,
+	}
+
+	/// Exposed trait-generic origin type.
+	#[pallet::origin]
+	pub type Origin<T> = RawOrigin<<T as Config>::AccountId>;
+
+	/// The full account information for a particular account ID.
+	#[pallet::storage]
+	#[pallet::getter(fn account)]
+	pub type Account<T: Config> = StorageMap<
+		_,
+		Blake2_128Concat,
+		T::AccountId,
+		AccountInfo<T::Index, T::AccountData>,
+		ValueQuery,
+	>;
+
+	/// Total extrinsics count for the current block.
+	#[pallet::storage]
+	pub(super) type ExtrinsicCount<T: Config> = StorageValue<_, u32>;
+
+	/// The current weight for the block.
+	#[pallet::storage]
+	#[pallet::getter(fn block_weight)]
+	pub(super) type BlockWeight<T: Config> = StorageValue<_, ConsumedWeight, ValueQuery>;
+
+	/// Total length (in bytes) for all extrinsics put together, for the current block.
+	#[pallet::storage]
+	pub(super) type AllExtrinsicsLen<T: Config> = StorageValue<_, u32>;
+
+	/// Map of block numbers to block hashes.
+	#[pallet::storage]
+	#[pallet::getter(fn block_hash)]
+	pub type BlockHash<T: Config> =
+	StorageMap<_, Twox64Concat, T::BlockNumber, T::Hash, ValueQuery>;
```

Comment:
interesting formatting, isn't more correct with one more indent ?

Reply:
Yeah, a hangover from copying the result of the template